core/src/main/python/wlsdeploy/util/cla_utils.py

```diff
@@ -973,19 +979,64 @@ def is_boolean_switch(self, key):
         return key in self.BOOLEAN_SWITCHES
 
     def is_compare_model_output_dir_switch(self, key):
-        return self.COMPARE_MODEL_OUTPUT_DIR_SWITCH == key
+        return self.OUTPUT_DIR_SWITCH == key
 
     def _validate_compare_model_output_dir_arg(self, value):
         method_name = '_validate_compare_model_output_dir_arg'
         try:
             variables = JFileUtils.validateDirectoryName(value)
         except JIllegalArgumentException, iae:
-            ex = exception_helper.create_cla_exception('WLSDPLY-01637', value, iae.getLocalizedMessage(), error=iae)
+            ex = exception_helper.create_cla_exception('WLSDPLY-01616', value, iae.getLocalizedMessage(), error=iae)
             ex.setExitCode(self.ARG_VALIDATION_ERROR_EXIT_CODE)
             self._logger.throwing(ex, class_name=self._class_name, method_name=method_name)
             raise ex
         return variables.getAbsolutePath()
 
+    def is_target_switch(self, key):
+        return key == self.TARGET_SWITCH
+
+    def _validate_target_arg(self, value):
+        method_name = 'validate_kubernetes_script_file_switch'
+
+        # Check for allowed targets
+
+        allowed_values = [ 'k8s' ]
```

Comment:
Should any value be allowed here, as long as the directory exists? Not sure if we are allowing users to extend this and make their own? Opinions?

Reply:
Base on what we discussed on slack, we can allow different values

Reply:
done